File: Domain/Models/Appointment.cs
namespace Domain.Models
{
    //مواعيد
    public class Appointment
    {
        public Appointment()
        {
            Id = Guid.NewGuid();
        }

        public Guid Id { get; set; }
        public DateTime TheAppointment { get; set; }
        public bool IsDeleted { get; set; } = false;
        public bool IsBooked { get; set; } = false;
        public Guid ServiceId { get; set; }
        public Service Service { get; set; }
        public Guid? ReservationId { get; set; }
        public Reservation? Reservation { get; set; }
    }
}

File: ScheduledServicesCenterUI/src/models/appointment.ts

import { Guid } from 'guid-typescript'; 
import { Service } from './service';
import { Reservation } from './reservation';


export class Appointment {

    Id: Guid | undefined;
    TheAppointment: Date | undefined;
    IsDeleted: boolean | undefined;
    IsBooked: boolean | undefined;
    ServiceId: Guid | undefined;
    Service: Service | undefined;
    ReservationId: Guid | null | undefined;
    Reservation: Reservation| null | undefined;
}

